docs/dcp2_system_design.rst

```diff
@@ -1223,49 +1247,117 @@ When a data release first enters the preparation phase, it is identical to the
+
+Alterations via ``complete`` staging areas
+------------------------------------------
+
+|nn| Before the ``delta`` staging areas specification was written, the DCP
+utilized the functionality that ``normal`` staging areas can contain multiple
+versions of an entity and reuse them to import the updates to a project. This
+is still being supported for backwards compatibility. |ne|
+
+
+Alterations via ``update`` staging areas
+----------------------------------------
+
+An ``update`` staging area has the ``type`` property set to ``update``. An
+``update`` staging area facilitates the same alterations as a ``delta``
+staging area, albeit in a less space-efficient form, and potentially causing
+redundant work in the importer. It has the significant advantage of being
+easier to implement. An import of an ``update`` staging area may take longer
+and may be expensive for large datasets (e.g. Tabula Muris). In this cases a
+``delta`` staging area should be utilized.
+
+An ``update`` staging area contains a complete set of subgraphs, metadata
+entities, and descriptors. The ``entity_id`` of entities, and the
+``links_id`` of subgraphs, must be stable. Only genuinely new entities
+(or subgraphs) may be assigned a new ``entity_id`` (or ``links_id``). 
+
+When the partitioning of the graph into subgraphs changes, the staging area
+source should minimize ID churn: if the metadata graph is partitioned into
+100 subgraphs before an update, and 200 after, the source should write all
+200 subgraphs with a new ``links_id`` each. If instead the number of
+subgraphs remains the same but some or all of the subgraphs are subject to an
+update like the addition of an entity, the same ``links_id`` should be used
+for each updated subgraph. The ``links_id`` of unchanged subgraphs should
+remain the same, obviously. In between those two extremes (all new
+``links_id`` values or all the same), the source is free to pick any suitable
+algortithm for allocating new ``links_id`` values and reusing existing ones.
+
+Before importing an ``update`` staging area, the importer empties all entity
+and links tables in the *target* dataset. The *target* dataset is the TDR
+dataset into which the staging area is being imported. The importer does not
+delete any data files. It then proceeds importing the ``update`` staging area
+as if it were a ``complete`` staging area with one exception: If the importer
+encounters a descriptor referencing a data file that is absent from the
+staging area, the importer checks for an existing data file in the target
+dataset. If one exists and has the CRC32C or SHA-256 checksum specified in
+the descriptor, the importer continues to import the descriptor, referencing
+the pre-existing data file in the target dataset.
+
+Note that an ``update`` staging area may not alter the ``file_id``, ``size``,
+``crc32c`` and ``sha256`` properties, it may alter or add other properties
+such as ``file_name`` or ``describedBy``. If there is no existing data file
```

Comment:
In a following line (R1433, not modified by this PR) it is stated that renaming data files is not currently supported.

Does this still hold true? or is it only applicable to `delta` staging areas?

Reply:
Good point. 

@ncalvanese1, I recall adding that limitation based on a comment by @nmalfroy in an earlier PR. AFAICT, the limitation should apply here as well, or be removed completely. WDYT?

Reply:
This limitation should also apply here, yes.